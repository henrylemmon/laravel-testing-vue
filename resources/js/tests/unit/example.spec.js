import { mount, shallowMount } from '@vue/test-utils'
import ExampleComponent from '../../components/ExampleComponent.vue'

test('it works', () => {
    expect(1 + 1).toBe(2)
})

describe('ExampleComponent', () => {
    it('renders props.msg when passed', () => {
        const words = 'New Words'
        const shallowWrapper = shallowMount(ExampleComponent, {
            propsData: { words }
        })
        expect(shallowWrapper.text()).toMatch(words)
    })

    test('should mount without crashing', () => {
        const wrapper = mount(ExampleComponent)

        expect(wrapper).toMatchSnapshot()
    })
})

