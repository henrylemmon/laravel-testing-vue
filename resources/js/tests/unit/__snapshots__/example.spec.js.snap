// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExampleComponent should mount without crashing 1`] = `
<div class="container">
  <div class="row justify-content-center">
    <div class="col-md-8">
      <div class="card">
        <div class="card-header">Example Component</div>
        <div class="card-body">

        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ExampleComponent.vue should mount without crashing 1`] = `
<div class="container">
  <div class="row justify-content-center">
    <div class="col-md-8">
      <div class="card">
        <div class="card-header">Example Component</div>
        <div class="card-body">

        </div>
      </div>
    </div>
  </div>
</div>
`;
